### ISSUE TITLE
[ECS Anywhere] Latest ecs-agent/ecs-init breaks compatibility, EBS mount error, v1.78.0 - v1.81.0

### ISSUE THREAD

**Issue by frehov** [aws/amazon-ecs-agent#4089]
### Summary

Trying to upgrade to latest version 1.81.0, breaks our install on an external instance.

We will be rolling back and pinning to before 1.78.0.

### Description

During an upgrade attempt of our docker version and ecs-agent version, our ssm-documents kept failing.
Running the upgrade manually according to the [ecs-anywhere upgrade guide](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-anywhere-updates.html) is kept failing to start, and inspecting the logs, it had started looping during the PreStart step. with the logs posted below.

Going through the history, this seems to have been introduced from version 1.78.0 with from the inclusion of EBS task volumes, https://github.com/aws/amazon-ecs-agent/commit/b9d174a8b602d436aa0c912bd65860f42f3d7103.

We believe this should be toggleable, or should be hidden behind `config.runningInExternal` at this point in the engine init https://github.com/aws/amazon-ecs-agent/blob/master/ecs-init/engine/engine.go#L146 as it currently is breaking ecs-anywhere (external) instances.

edit: our /mnt folder on the hosts are read-only and cannot be modified, I believe this presents a missing testcase and thus not covered by the testing done in https://github.com/aws/amazon-ecs-agent/pull/3952 which mentions testing with ecs-anywhere-install.sh

### Expected Behavior
Upgrade/installs successfully, and starts up without issue

### Observed Behavior
Upgrades the package, and fails during PreStart of the service with:
`could not create EBS mount directory: mkdir /mnt/ecs: read-only file system`

### Environment Details

Docker info:  
```
Client: Docker Engine - Community
 Version:    25.0.3
 
Server:
 Containers: 2
  Running: 0
  Paused: 0
  Stopped: 2
 Images: 5
 Server Version: 25.0.3
 Storage Driver: overlay2
  Backing Filesystem: xfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
 Kernel Version: 4.18.0-513.5.1.el8_9.x86_64
 Operating System: Red Hat Enterprise Linux 8.9 (Ootpa)
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.35GiB
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

### Supporting Log Snippets

2024-02-07T14:01:03Z [INFO] pre-start
2024-02-07T14:01:03Z [INFO] pre-start: enabling loopback routing
2024-02-07T14:01:03Z [INFO] pre-start: disabling ipv6 router advertisements
2024-02-07T14:01:03Z [INFO] pre-start: creating credentials proxy route
2024-02-07T14:01:03Z [ERROR] could not create EBS mount directory: mkdir /mnt/ecs: read-only file system
2024-02-07T14:01:04Z [INFO] post-stop
2024-02-07T14:01:04Z [INFO] Cleaning up the credentials endpoint setup for Amazon Elastic Container Service Agent

**Comment by frehov** [comment]
Would also be nice if someone could confirm the version number syntax on the option `--ecs-version`.

I have currently tried the following options for `--ecs-version` switch:
- v1.77
- v1.77-1
- v1.77.0
- v1.77.0-1
- 1.77
- 1.77-1
- 1.77.0
- 1.77.0-1 (this one finally worked)

I had to look at the installed package to see the expected version number format

**Comment by prateekchaudhry** [comment]
Hi @frehov , thanks for reporting this issue. We have merged a fix which should be expected to be available when next ECS agent.
Meanwhile, please use the mitigation to use ecs-init 1.77.0 package when installing ecs packages
`bash /tmp/ecs-anywhere-install.sh --ecs-version 1.77.0-1 --region ....`

**Comment by prateekchaudhry** [comment]
The fix has been released with agent [1.82.0](https://github.com/aws/amazon-ecs-agent/tree/v1.82.0). Recommend to use latest available agent. Closing this.
